source/byte_buf.c

```diff
@@ -89,26 +114,25 @@ int aws_byte_buf_init_copy_from_cursor(
     struct aws_byte_buf *dest,
     struct aws_allocator *allocator,
     struct aws_byte_cursor src) {
-    assert(allocator);
-    assert(dest);
-
-    dest->len = 0;
-    dest->capacity = 0;
-    dest->allocator = NULL;
-    if (src.ptr == NULL) {
-        dest->buffer = NULL;
-        return AWS_OP_SUCCESS;
+    if (!allocator || !dest || !aws_byte_cursor_is_valid(&src)) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
     }
 
-    dest->buffer = (uint8_t *)aws_mem_acquire(allocator, sizeof(uint8_t) * src.len);
+    /* clear the dest before updating it. */
+    *dest = (struct aws_byte_buf){0};
+
+    dest->buffer = (uint8_t *)aws_mem_acquire(allocator, src.len);
     if (dest->buffer == NULL) {
+        AWS_POSTCONDITION(aws_byte_cursor_is_valid(&src));
```

Comment:
Not sure this is worth doing when src is passed by value.

Reply:
Nice catch.